universe/manifest.tt

```diff
@@ -0,0 +1,40 @@
+class @vikram.hello
+#_[thingpedia_name="Test 1"]
+#_[thingpedia_description="Test 1 for Movies"]
+#[license="BSD-3-Clause"]
+#[license_gplcompatible=true]
+#[subcategory="media"]
+{
+    import loader from @org.thingpedia.v2();
+    import config from @org.thingpedia.config.none();
+
+    entity movie #_[description = "Movie Information"];
+    entity actor #_[description = "People in movie"];
+    monitorable query movie(in req query : String
+                                    #_[canonical={
+                                        base = ["movie", "title"],
+                                        passive_verb = ["titled #", "starring #", "named #"],
+                                    }],
```

Comment:
Rather that using an input parameter to search a movie by name, you should use filters and query hints to search by name. That is, you would represent the command:
```
find me Titanic
```
as
```
@com.tmbd.movie(), id =~ "titanic";
```

Inside the index.js, the third parameter to the query function is `hints`, an object with the [CompiledQueryHints](https://almond-static.stanford.edu/jsdoc/thingtalk/latest/interfaces/runtime.compiledqueryhints.html) interface. `hints.filter` will contain the information to search by name.

By making this change, you'll be able to search movies by other characteristics rather than by name:
- by actor: "search movies with Leonardo di Caprio"
- by date: "search movies from 1993" (and also, "what's the latest movie")
- by official rating: "search PG-13 movies"
- by quality: "what's the best rated movie of 2020"
- even by nothing: "recommend me a movie"